File: EduSpace_BE/account-service/src/main/java/com/eduspace/accountservice/model/dto/request/LoginRequest.java
package com.eduspace.accountservice.model.dto.request;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import lombok.AccessLevel;
import lombok.Data;
import lombok.experimental.FieldDefaults;

@Data
@FieldDefaults(level = AccessLevel.PRIVATE)
public class LoginRequest {

    @Email
    @NotBlank
    String email;

    @NotBlank
    String password;

    String otp; // Optional TOTP code
}

File: eduspace_mobile_v2/lib/data/models/request/auth/login_request.dart
class LoginRequest {
  final String email;
  final String password;
  final String? otp;

  LoginRequest({
    required this.email,
    required this.password,
    this.otp,
  });

  Map<String, dynamic> toJson() {
    return {
      'email': email,
      'password': password,
      if (otp != null) 'otp': otp,
    };
  }
}
